experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -743,6 +743,103 @@ describe("SharedTree", () => {
 			unsubscribe1();
 			unsubscribe2();
 		});
+
+		describe("can concurrently restore and edit removed tree", () => {
+			const sb = new SchemaBuilder({ scope: "shared tree undo tests" });
+			const schema = sb.intoSchema(sb.list(sb.list(sb.string)));
+
+			it("with the edit sequenced first", () => {
+				const provider = new TestTreeProviderLite(2);
+				const content = {
+					schema,
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: [["a"], ["b"]] as any,
+				} satisfies InitializeAndSchematizeConfiguration;
+				const tree1 = provider.trees[0].schematizeInternal(content);
+				const { undoStack: undoStack1, unsubscribe: unsubscribe1 } =
+					createTestUndoRedoStacks(tree1.checkout);
+				const tree2 = provider.trees[1].schematizeInternal(content);
+				const { undoStack: undoStack2, unsubscribe: unsubscribe2 } =
+					createTestUndoRedoStacks(tree2.checkout);
+
+				provider.processMessages();
+
+				// Validate insertion
+				validateTreeContent(tree2.checkout, content);
+
+				// edit subtree
+				tree2.root[0].insertAtEnd(["c"]);
+				provider.processMessages();
+				assert.equal(tree1.root[0][0], "a");
+				assert.equal(tree1.root[0][1], "c");
```

Comment:
Can we use `validateTreeContent` so that it's a little easier to get a picture of the current state?
And if you do that, can you do away with the `["b"]` part of the data?